packages/router/src/Set.tsx

```diff
@@ -35,38 +61,61 @@ export function Set<WrapperProps>(props: SetProps<WrapperProps>) {
     const path = route.props.path as string
 
     const { match } = matchPath(path, location.pathname, routerState.paramTypes)
-    if (match) {
-      if (isPrivate && unauthorized()) {
-        if (loading) {
-          return route.props?.whileLoading?.() || null
-        } else {
-          const currentLocation =
-            global.location.pathname +
-            encodeURIComponent(global.location.search)
-
-          const unauthenticatedRoute = routerState.routes.filter(
-            ({ name }) => unauthenticated === name
-          )[0]
-
-          return (
-            <Redirect
-              to={`${unauthenticatedRoute.path}?redirectTo=${currentLocation}`}
-            />
-          )
-        }
-      }
+    if (!match) {
+      continue
+    }
+
+    if (privateSet && unauthorized()) {
+      if (loading) {
+        return route.props?.whileLoading?.() || null
+      } else {
+        const currentLocation =
+          global.location.pathname + encodeURIComponent(global.location.search)
+
+        const unauthenticatedPath = routerState.routes.filter(
+          ({ name }) => unauthenticated === name
+        )[0]?.path || '/'
```

Comment:
I'm wondering if we shouldn't throw when we can't find the unauthenticated route. I know our types do not make that a requirement, but I'm thinking redirecting to `/` may be unexpected, and in the case of a typo it might be nice to know:

"We could not find a route named `Login`"

Reply:
I can throw an error if `unauthenticated` is provided, but I can't find it. 
What do you want to do if `unauthenticated` isn't provided at all? (i.e. `unauthenticated === undefined`)

Reply:
I think we should throw in that case too and say something like: "You have specified a Set should be private, but does not contain an authenticated route."

Reply:
I added two different throws, please have a look